File: Time on Task.py
# 2013 J4

chore_count = 0
max_time = int(input())
chore_times = []
sum_chore_times = 0

for i in range(int(input())):
    chore_times.append(int(input()))

# [5,4,3,2,1]

for i in chore_times:
    if i >= max_time:
        chore_times.remove(i)

# [1,2,3,4,5]

chore_times.sort()

for i in chore_times:
    sum_chore_times += i
    if sum_chore_times <= max_time:
        chore_count += 1


print(chore_count)

File: C++/2013/Time on Task.cpp
// J2

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int chore_count = 0, sum_chore_times = 0, max_time, chores;
    vector<int> chore_times;

    cin >> max_time;
    cin >> chores;

    // get input
    for (size_t i = 0; i < chores; i++)
    {
        int num;
        cin >> num;
        chore_times.push_back(num);
    }

    // filter input if bigger than max time
    for (size_t i = 0; i < chore_times.size(); i++)
    {
        if(chore_times[i] > max_time) chore_times.erase(chore_times.begin() + i);
    }

    sort(chore_times.begin(), chore_times.end());

    for (size_t i = 0; i < chore_times.size(); i++)
    {
        sum_chore_times += chore_times[i];
        if (sum_chore_times <= max_time)
        {
            chore_count++;
        }
        
    }

    cout << chore_count << endl;
        

    
    
}